### ISSUE TITLE
Preview sometimes not loading correctly

### ISSUE THREAD

**Issue by shirnschall** [cerebroapp/cerebro#285]
- [x] I am on the [latest](https://github.com/KELiON/cerebro/releases/latest) Cerebro.app version
- [x] I have searched the [issues](https://github.com/KELiON/cerebro/issues) of this repo and believe that this is not a duplicate

- **OS version and name**: ubuntu gnome 16.04.2 LTS (Xenial Xerus)
- **Cerebro.app version**: 0.2.8
- **Relevant information from devtools** _(See above how to open it)_: N/A

## Issue
As you can see in the image sometimes the preview is not loading correctly. After pressing backspace a few times and trying some other keys it usually begins to load the preview. Sometimes the preview works and sometimes it does not. When it is working it sometimes stops working the next time I open the search bar.
![cerebro](https://cloud.githubusercontent.com/assets/11620061/24923405/b4dfea96-1ef1-11e7-9fb7-0d629e39fd94.png)

**Comment by KELiON** [comment]
@hirnschallsebastian interesting bug! Is it possible for you to record video/gif?

**Comment by hirnschallsebastian** [comment]
There you go... I had to zip it since github does not support video. 
When the searchbar is empty in the video - between searches - I pressed backspace several times...(you can't see it in the video since the bar is already empty)

[cerebro-bug.zip](https://github.com/KELiON/cerebro/files/916114/cerebro-bug.zip)

**Comment by hirnschallsebastian** [comment]
I converted it to a gif so you do not have to download it.
![ezgif-3-5e9008278a](https://cloud.githubusercontent.com/assets/11620061/24952364/7f4b1c26-1f77-11e7-8d07-bd81745f3bc3.gif)

**Comment by maximbaz** [comment]
Just a thought, try quitting cerebro and starting it again as `$ cerebro --disable-gpu` (or vice versa if you are already running it with this flag).

**Comment by hirnschallsebastian** [comment]
No unfortunately it is still not working correctly

**Comment by KELiON** [comment]
Did you start Cerebro recently or you've been using it for a while? So, my question is did you have these kind of problems on previous versions? For me it looks like bug of electron, not Cerebro (because this area is not transparent in application), so would be great if you can check some of previous versions (i.e. [0.2.2](https://github.com/KELiON/cerebro/releases/tag/0.2.2)).

**Comment by hirnschallsebastian** [comment]
I have just started using your application recently. It all seems to work fine in v.0.2.2 though.

**Comment by Enzime** [comment]
Could you try doing a `git bisect`?

If you haven't already got a copy of the repository just follow these steps:

    git clone https://github.com/KELiON/cerebro.git
    cd cerebro

This will create a copy of the repository into the `cerebro` folder.

To actually use `git bisect`:

    git bisect start
    git bisect good 0.2.2
    git bisect bad

After you've selected the first good commit and first bad commit, now you want to repeat these steps until git bisect outputs what your bad commit is (e.g. `d18417c6f3f57620fae4971c3b4f10c42c393df3 is the first bad commit`)

1. Install all the dependencies for that specific commit then run Cerebro, give it a little while to finish compiling everything, then test whether or not you get that bug in this specific version. After you've finished testing, just switch to the terminal you were running this in and press `Ctrl-C`

    yarn && cd app && yarn && cd .. && yarn run dev

2. First cleanup the folder then mark the commit as either `good` or `bad` using `git bisect` depending on whether it had the bug or not

    git reset --hard
    git bisect good

Once you've reached the end, you can run `git bisect log` and copy paste that output here :+1:

**Comment by glja021** [comment]
I just stumbled across this bug as well, using `version 0.2.8`, I'm running Ubuntu Gnome 16.10.

Looks like it randomly fails to display the results, clearing the input and retyping anything seems to fix it.
Seems like a visual glitch because it's still usable while invisible :stuck_out_tongue:

**Comment by hirnschallsebastian** [comment]
looks like electron 1.6.2 is not working right :/

```
git bisect start
# good: [34b74adc90ff1e5e3137ba3f25bb9e478ea620d6] Bump version to 0.2.2
git bisect good 34b74adc90ff1e5e3137ba3f25bb9e478ea620d6
# bad: [3567773f2d5c06567dd3d631ed7fa7c3e797558f] Merge pull request #276 from SkaceKamen/basicapps-preview-fix
git bisect bad 3567773f2d5c06567dd3d631ed7fa7c3e797558f
# good: [afc921d5b4815332cd29e3eb0ddf99d71a1d7d22] Merge pull request #189 from tenorz007/patch-1
git bisect good afc921d5b4815332cd29e3eb0ddf99d71a1d7d22
# good: [afc921d5b4815332cd29e3eb0ddf99d71a1d7d22] Merge pull request #189 from tenorz007/patch-1
git bisect good afc921d5b4815332cd29e3eb0ddf99d71a1d7d22
# good: [c3937fb7cc2782de635de99984c80d6e0d7aa9d0] Make config.json nicely formatted
git bisect good c3937fb7cc2782de635de99984c80d6e0d7aa9d0
# bad: [f24691db4347dfcb744df8beefe6290252b3f23f] Merge pull request #263 from Enzime/feature/floating-on-linux
git bisect bad f24691db4347dfcb744df8beefe6290252b3f23f
# good: [7d9c4b0d0043f7763bde4ecf2fbb8d78e8ff3bbc] Merge pull request #252 from lawgsy/patch-1
git bisect good 7d9c4b0d0043f7763bde4ecf2fbb8d78e8ff3bbc
# bad: [2fe3ca3e3bab53745ff07ef0782f7fb60bf73e9e] Merge pull request #254 from KELiON/feature/donate
git bisect bad 2fe3ca3e3bab53745ff07ef0782f7fb60bf73e9e
# good: [51b12fa2f0b84da32685143c4daf61366cae3015] Merge pull request #255 from KELiON/feature/lightweight-vim-mode
git bisect good 51b12fa2f0b84da32685143c4daf61366cae3015
# bad: [1af46fa77a3348ad8ccad83497281f13b760cdf9] Update electron to version 1.6.2
git bisect bad 1af46fa77a3348ad8ccad83497281f13b760cdf9
# bad: [1af46fa77a3348ad8ccad83497281f13b760cdf9] Update electron to version 1.6.2
git bisect bad 1af46fa77a3348ad8ccad83497281f13b760cdf9
# good: [f69cbbc7ef30ae086185fe06fe205e7b3bdf80df] Merge pull request #247 from maximbaz/prettify-config-json
git bisect good f69cbbc7ef30ae086185fe06fe205e7b3bdf80df
# good: [fb6dc9a62dd65ee0b04b3eb468c19a1768a6f84a] Added donate dialog and item in tray menu
git bisect good fb6dc9a62dd65ee0b04b3eb468c19a1768a6f84a
# first bad commit: [1af46fa77a3348ad8ccad83497281f13b760cdf9] Update electron to version 1.6.2
```

**Comment by maximbaz** [comment]
It's cool that you narrowed the search down by doing a git bisect! Now we know the breaking change probably occurred in electron v1.5 or v1.6. 

Quick search revealed https://github.com/electron/electron/issues/8505, I'm not saying that this is exactly our case, but it is at least an example of where rendering broke after upgrade to v1.5.

**Comment by hirnschallsebastian** [comment]
I just want this awesome tool to work on my linux box :)
Thanks @Enzime for telling me about git bisect. It's super cool and I am sure I can use it with my own projects in the future!

**Comment by guemidiborhane** [comment]
how can i fix the problem ?

**Comment by KELiON** [comment]
@guemidiborhane we will update electron to the latest version and probably it will fix the problem

**Comment by guemidiborhane** [comment]
@KELiON i really like it i was using @albertlauncher and this is the only issue i'm having that is preventing me to fully migrate :disappointed:

**Comment by danilopinotti** [comment]
I'm also having this problem.
OBS: I use exactly the same OS

**Comment by dmeijboom** [comment]
I have the same problem. I'm running Ubuntu 17.04 and installed the latest debian package. My graphic card is nvidia.

**Comment by Taskkill** [comment]
I have same problem, Debian 9. I have gforce m840 but system doesn't use it as gpu, by default it's using bult in intel hd graphic.

**Comment by emanjavacas** [comment]
same here with ubuntu 16.04, gnome (using Intel HD Graphics 520)

**Comment by IgnusG** [comment]
Same bug on Arch, Deepin using Intel Graphics or Radeon. Happens almost every second time :/

**Comment by Strus** [comment]
Same error on newest Elementary OS.

**Comment by marcelogdeandrade** [comment]
Same error on newest Elementary OS

**Comment by tfmiranda** [comment]
Same error on newest Deepin OS using Radeon card.

**Comment by floringogianu** [comment]
Same on Ubuntu `17.04`, Budgie desktop environment, cerebro `0.3.1`

**Comment by mokyle** [comment]
Cerebro v0.2.7.deb do not have this problem

**Comment by omouren** [comment]
Same on `Antergos`, Cerebro `0.3.1`

**Comment by JozkoDebugger** [comment]
Same bug happens on **Linux Mint 17.3** with Cinnamon 2.8.8.
GeForce 460M
**Cerebro 0.3.1 AppImage**

**Comment by dc3671** [comment]
Same on Arch, KDE desktop. Cerebro 0.3.1

**Comment by lucas-china** [comment]
Same on Deepin 15.4, cerebro_0.3.1_amd64.deb

**Comment by scmanjarrez** [comment]
Version 0.3.2 still suffers from this bug.
Ubuntu 16.04, x64, Gnome-Shell 3.20.

**Comment by elissongois** [comment]
How can I solve this?
A JavaScript error occurred in the main process

Uncaught Exception:
TypeError: visitor(...).screenview is not a function
    at module.exports.exports.screenView.screenName (/opt/Cerebro/resources/app.asar/main.js:251:15)
    at module.exports.exports.default (/opt/Cerebro/resources/app.asar/main.js:915:30)
    at App.module.exports._electron.app.on (/opt/Cerebro/resources/app.asar/main.js:653:43)
    at emitTwo (events.js:111:20)
    at App.emit (events.js:191:7)
Uncaught Exception:
TypeError: Cannot read property 'webContents' of undefined
    at EventEmitter.module.exports._electron.ipcMain.on (/opt/Cerebro/resources/app.asar/main.js:687:47)
    at emitTwo (events.js:106:13)
    at EventEmitter.emit (events.js:191:7)
    at WebContents.<anonymous> (/opt/Cerebro/resources/electron.asar/browser/api/web-contents.js:247:37)
    at emitTwo (events.js:106:13)
    at WebContents.emit (events.js:191:7)

**Comment by mxdlzg** [comment]
Same on Deepin 15.5.
cerebro 3.1/3.2

**Comment by Otoru** [comment]
Same on Deepin 15.5, cerebro-0.3.1-x86_64.AppImage.

**Comment by dubisdev** [comment]
Hi all! I'm closing this issue as outdated (+5 years)

We have just released Cerebro `v0.9.0`.

Give it a try and, if the problem persists, feel free to open a new issue in the repo.
